src/test/java/com/synopsys/integration/blackduck/configuration/BlackDuckServerConfigBuilderTest.java

```diff
@@ -140,12 +181,366 @@ public void testPopulatingExecutorService() throws Exception {
             executorServiceField.setAccessible(true);
             assertTrue(executorServiceField.get(blackDuckServerConfig) instanceof NoThreadExecutorService);
 
+            assertNotNull(blackDuckServerConfigBuilder.getExecutorService());
             blackDuckServerConfigBuilder.setExecutorService(executorService);
+            assertEquals(executorService, blackDuckServerConfigBuilder.getExecutorService().orElse(null));
             blackDuckServerConfig = blackDuckServerConfigBuilder.build();
             assertNotNull(executorServiceField.get(blackDuckServerConfig));
         } finally {
+            assert executorService != null;
```

Comment:
Why is this a different type of assertion?

Reply:
shutdownNow(), called on the next line, can throw NPE. Would you say to change this to an if check, or just remove the check completely?

Reply:
My confusion was in it not being ‘Assertions.assertNull’, but if this is for the test cleanup it’s fine